lib/cassy/cas.rb

```diff
@@ -158,14 +162,20 @@ def base_service_url(full_service_url)
     def detect_ticketing_service(service)
       # try to find the service in the valid_services list
       # if loosely_matched_services is true, try to match the base url of the service to one in the valid_services list
+      # if match_on_top_level_domain is true, ensure the base url of the service belongs to an allowed top level domain
```

Comment:
I think top level domain is the wrong terminology here. top level domains are like "com", "org", "ca", etc. "backupify.com" would be a second-level domain. For certs and DNS people will also talk about "wildcards"